src/tribler/core/components/libtorrent/tests/test_torrent_def.py

```diff
@@ -225,6 +225,9 @@ def test_torrent_no_metainfo():
     assert not tdef.is_private()
     assert tdef.get_name_utf8() == "video.avi"
     assert tdef.get_nr_pieces() == 0
+    assert tdef.torrent_info is None
+    tdef.load_torrent_info()
+    assert tdef.torrent_info is None
```

Comment:
Since `TorrentDefNoMetainfo` inherits from `TorrentDef`, calling `tdef.save()` on line 231 or 234 raises `IndexError: list index out of range`.